src/app/public/testing/resources-test.service.ts

```diff
@@ -38,7 +38,7 @@ export class SkyAppResourcesTestService {
     // that has no resource files from erroring during a unit test run unless it
     // explicitly tries to reference a resource string.
     let resourcesContext;
-    if (ROOT_DIR.indexOf('public') === -1) {
+    if (/(\/|\\)src(\/|\\)app(\/|\\)public(\/|\\)/.test(ROOT_DIR) === false) {
```

Comment:
Check for "/src/app/public/" instead of just "public" so that applications that include "public" in their name don't throw false positives.